pkg/api/error.go

```diff
@@ -109,6 +123,20 @@ func NewCloudError(statusCode int, code, target, message string, a ...interface{
 	}
 }
 
+// NewCloudError returns a new CloudError with category and dependency
+func NewCloudErrorWithCategory(statusCode int, code, target, message string, category CloudErrorCategory, dependency string, a ...interface{}) *CloudError {
```

Comment:
Where is this function used? We have a policy: "No dead code in the repo."
If it is used and will be referenced in the followup PR would you please add it to the followup?
Thank you.

Reply:
Removed and will place in follow-up PR.